obspy/signal/spectral_estimation.py

```diff
@@ -501,14 +619,17 @@ def __check_time_present(self, utcdatetime):
         would result in an overlap of the ppsd data base, False if it is OK to
         insert this piece of data.
         """
-        index1 = bisect.bisect_left(self._times_used, utcdatetime.timestamp)
-        index2 = bisect.bisect_right(self._times_used,
+        index1 = bisect.bisect_left(self._times_processed,
+                                    utcdatetime.timestamp)
+        index2 = bisect.bisect_right(self._times_processed,
                                      utcdatetime.timestamp + self.ppsd_length)
         if index1 != index2:
             return True
         else:
             return False
 
+    @deprecated("Support for old pickled PPSD objects will be dropped. "
+                "Please use new 'save_npz'/'load_npz' mechanism.")
```

Comment:
Yeah, I agree, not needed here, already moved this tiny check into the deprecated `load` routine in soon to be merged PR: https://github.com/obspy/obspy/pull/1166/files#diff-9b9d51c398e8720552f8a521d3f6c69bL631